src/include/migraphx/op/pad.hpp

```diff
@@ -51,6 +51,8 @@ struct pad
         return s;
     }
 
+    std::size_t pad_ndims() const { return pads.size() / 2; }
+
```

Comment:
We need a check to ensure pads.size() is an even number. 
Or we can separate the start and end pads of each dimension, then it is more readable when the graph is print out.